core/src/plan/primary_key.rs

```diff
@@ -101,113 +195,407 @@ impl<'a> PrimaryKeyPlanner<'a> {
     fn expr(
         &self,
         outer_context: Option<Rc<Context<'a>>>,
-        current_context: Option<Rc<Context<'a>>>,
+        current_context: &Rc<Context<'a>>,
         expr: Expr,
     ) -> PrimaryKey {
-        let check_primary_key = |key: &Expr| {
-            let key = match key {
-                Expr::Identifier(ident) => ident,
-                Expr::CompoundIdentifier { ident, .. } => ident,
-                _ => return false,
+        enum PossibleResults {
+            Matched(PrimaryKey),
+            Retry(Box<Expr>, Box<Expr>),
+        }
```

Comment:
ok, now I am beginning to find some points.
here, we currently add Partial variant to PrimaryKey enum.
rather than adding `PrimaryKey::Partial`, could we consider to add an extra param for partial primary key expression to `fn expr(..)`? then we might not need `PrimaryKey::Partial` and it may simplify the code logic.
I wonder your opinion. currently I'm not completely sure yet either.

Reply:
The issue is that it is a result of the expr function, not a parameter of the function.

Reply:
expr function is defined only for primary key planner. we don't need to be limited to the existing shape of the codes. then could we find something on it?

Reply:
Since the method is recursive, as it is appropriate given that the task has a recursive definition, we need a way to uniquely represent when a step of the recursion yields a partial result. Other than adding an enum option that represents it, which is the current solution, I have no other ideas.

Reply:
ok, lets keep talking on the comment below, there is the point I see. 
https://github.com/gluesql/gluesql/pull/1515#discussion_r1703113335